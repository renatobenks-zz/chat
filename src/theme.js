// @flow

import logo from './static/svg/logo/logo.svg';

const theme = {
  palette: {
    primary: '#24E28B',
    secondary: '#1FC6C1',
    gray: '#828796',
    white: '#fff',
    black: '#000',
    yellow: '#ffff00',
  },
  responsive: {
    extraSmall: {
      type: 'max-width',
      unit: '576px',
    },
    small: {
      type: 'min-width',
      unit: '576px',
    },
    medium: {
      type: 'min-width',
      unit: '768px',
    },
    large: {
      type: 'min-width',
      unit: '992px',
    },
    extraLarge: {
      type: 'min-width',
      unit: '1200px',
    },
  },
  svg: {
    logo,
  },
  padding: '1rem',
  header: {
    height: '70px',
  },
  avatar: {
    size: '4rem',
  },
};

export default theme;
