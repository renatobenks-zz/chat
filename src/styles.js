// @flow

import { css } from 'styled-components';

import theme from './theme';

type MediaType = {
  extraSmall: Function,
  small: Function,
  medium: Function,
  large: Function,
  extraLarge: Function,
};

export const media: MediaType = Object.keys(theme.responsive).reduce(
  (accumulator, currentValue) => {
    const { type, unit } = theme.responsive[currentValue];

    return {
      ...accumulator,
      [currentValue]: (...style) => css`
        @media (${type}: ${unit}) {
          ${css(...style)}
        }
      `,
    };
  },
  {
    extraSmall: style => style,
    small: style => style,
    medium: style => style,
    large: style => style,
    extraLarge: style => style,
  },
);
