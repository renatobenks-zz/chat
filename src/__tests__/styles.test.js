import { media } from '../styles';
import theme from '../theme';

const medias = Object.keys(media);

it('media unit types', () => {
  expect(medias).toMatchSnapshot();
  expect(theme.responsive).toMatchSnapshot();
});
