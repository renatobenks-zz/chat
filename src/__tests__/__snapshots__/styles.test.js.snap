// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`media unit types 1`] = `
Array [
  "extraSmall",
  "small",
  "medium",
  "large",
  "extraLarge",
]
`;

exports[`media unit types 2`] = `
Object {
  "extraLarge": Object {
    "type": "min-width",
    "unit": "1200px",
  },
  "extraSmall": Object {
    "type": "max-width",
    "unit": "576px",
  },
  "large": Object {
    "type": "min-width",
    "unit": "992px",
  },
  "medium": Object {
    "type": "min-width",
    "unit": "768px",
  },
  "small": Object {
    "type": "min-width",
    "unit": "576px",
  },
}
`;
